packages/server/src/scripts/import_arpa_reporter_dump.js

```diff
@@ -0,0 +1,469 @@
+        )
+    );
+    let inserted = await knex("agencies")
+        .insert(agenciesToCreate)
+        .returning("*");
+    const idLookup = _.chain(inserted)
+        .map((insertedRow, idx) => [
+            dbContents.agencies[idx].id,
+            insertedRow.id,
+        ])
+        .fromPairs()
+        .value();
+
+    // Then update main_agency_id on tenant rows
+    for (const tenant of insertedRowsByTable.tenants) {
+        const mainAgencyId = idLookup[tenant.__unmappedFutureMainAgencyId];
+        tenant.main_agency_id = mainAgencyId;
+        await knex("tenants")
+            .where("id", tenant.id)
+            .update({ main_agency_id: mainAgencyId });
+    }
+
+    // Then update any non-main agencies to be parented by their tenant's main agency (since
+    // standalone ARPA Reporter did not have a concept of parent agencies)
+    const agencyIds = _.map(inserted, "id");
+    await knex.raw(
+        `
+        UPDATE agencies
+        SET
+            parent = tenants.main_agency_id,
+            main_agency_id = tenants.main_agency_id,
+        FROM tenants
+        WHERE
+            agencies.tenant_id = tenants.id
+            AND agencies.id in :agencyIds
+        `,
+        { agencyIds }
+    );
+    inserted = await knex("agencies").select("*").whereIn("id", agencyIds);
+
+    return { inserted, idLookup };
+}
+
+async function importUsers(dbContents, idLookupByTable, insertedRowsByTable) {
+    const roles = await knex('roles').select('*');
+    const adminRole = roles.find(r => r.name == 'admin').id;
+    const staffRole = roles.find(r => r.name == 'staff').id;
+
+    const mainAgencyByTenant = _.chain(insertedRowsByTable.tenants)
+        .keyBy("id")
+        .mapValues("main_agency_id")
+        .value();
+    const usersToCreate = dbContents.users.map((user) =>
+        rekeyForeignKeys(
+            {
+                email: user.email,
+                name: user.name,
+                // Note: ARPA Reporter had a "reporter" role. For our purposes,
+                // anything non-admin becomes "staff"
+                role_id: user.role === 'admin' ? adminRole : staffRole,
+                tenant_id: user.tenant_id,
+                // Copied users belong to the main agency of their tenant
+                agency_id:
+                    mainAgencyByTenant[idLookupByTable.tenants[user.tenant_id]],
+            },
+            idLookupByTable,
+            ["agency_id"]
+        )
+    );
+
+    // Check if there are any existing users with the specified emails
+    const usersWithDupeEmails = await knex("users")
+        .select("email")
+        .whereIn("email", _.map(usersToCreate, "email"));
+    if (usersWithDupeEmails.length != 0) {
+        console.error(
+            "Found duplicate emails",
+            _.map(usersWithDupeEmails, "email")
+        );
+        throw new Error(
+            "found users with duplicate emails! Delete them first, or update script to handle."
```

Comment:
There are definitely duplicates in the tables, especially for the USDR employees/volunteers that show up in most of the dbs, so we'll definitely need to address this before we can run it

Reply:
Oh, good point -- I didn't think about our own users on partners' instances.

We can probably have the script prompt for a remapped email and use a plus-address (at least for usdigitalresponse.org and other Gmail addresses). Or we could omit such users, but there might be rows in other tables that refer to them (e.g. `validated_by`).